src/common/utils/pagination.ts

```diff
@@ -0,0 +1,41 @@
+import { DeserializedList, PaginationOptions} from '../interfaces';
+// Autopaginatable class definition
+export class Autopaginatable<T> {
```

Comment:
I feel this should be `AutoPaginatable`. Especially since the function is `autoPagination` later.